File: leet_119_pascal_triangle_ii.cpp
class Solution {
public:
    vector<int> getRow(int row)
    {
        if (!row)
            return {1} ;
        if (row == 1)
            return {1, 1};

        vector<vector<int>> p(row + 1, vector<int>()) ;
        p[0] = {1} ;
        p[1] = {1, 1};
        int i = 1;
        while (++ i < row + 1)
        {
            int j = -1 ;
            while (++ j < i + 1)
            {
                if (!j || j == i)
                    p[i].push_back(1) ;
                else
                    p[i].push_back(p[i - 1][j - 1] + p[i - 1][j]) ;
            }
        }
        return (p[row]) ;
    }
};

File: leet_119_pascal_triangle_ii.py
class Solution:
    def getRow(self, row: int) -> List[int]:
        if row is 0:
            return [1]
        if row is 1:
            return [1, 1]
        p = [[] for _ in range(row + 1)]
        p[0] = [1]
        p[1] = [1, 1]
        for i in range(2, row + 1):
            for j in range(i + 1) :
                if j is 0 or j is i:
                    p[i].append(1)
                else:
                    p[i].append(p[i - 1][j - 1] + p[i - 1][j])
        return p[row]
